### PR TITLE
init lambda

### PR DESCRIPTION
**Pull request by anderysenuta**
**Service**
- https://hls6p04919.execute-api.eu-west-1.amazonaws.com/dev/products
- https://hls6p04919.execute-api.eu-west-1.amazonaws.com/dev/products/1669af17-d287-5094-b005-4b143441442f

**Error example:**
- https://hls6p04919.execute-api.eu-west-1.amazonaws.com/dev/products/xxxxx 

**Additional tasks**

- [x] Async/await is used in lambda functions (see test files)
- [x] ES6 modules are used for product-service implementation
- [x] Webpack is configured for product-service
- [x] SWAGGER documentation (https://app.swaggerhub.com/apis-docs/mtg3/mtg/1.0.0)
- [x] Lambda handlers are covered by basic UNIT tests
- [x] Lambda handlers (getProductsList, getProductsById) code is written not in 1 single module (file) and separated in codebase.- [ ] Main error scenarious are handled by API ("Product not found" error, try catch blocks are used in lambda handlers).

FrontEnd PR link - https://github.com/anderysenuta/nodejs-aws-fe/pull/2

### CODE REVIEW

File: product-service/handlers/get-product-by-id/index.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { APIGatewayProxyHandler } from 'aws-lambda';
+import products from '../../mocks/products';
+
+export const handler: APIGatewayProxyHandler = async (event, _context) => {
```
Comment (Emerkof) [inline product-service/handlers/get-product-by-id/index.ts:4]: @anderysenuta Request to `/products/{id}` route suggests me downloading a `test.csv` file with the following content:

```csv
1,"blah",123
2,"qwe",456
```

No matter what `id` I type in, the response content is the same.

Could you fix it?

File: product-service/handlers/get-products/index.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { APIGatewayProxyHandler } from 'aws-lambda';
+import products from '../../mocks/products';
+
+export const handler: APIGatewayProxyHandler = async () => {
```
Comment (Emerkof) [inline product-service/handlers/get-products/index.ts:4]: Could you please provide an URL to your CloudFront-deployed Frontend application, so I can make sure this route is integrated into your Frontend app?

File: product-service/handlers/get-product-by-id/index.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  try {
```
Comment (Emerkof) [inline product-service/handlers/get-product-by-id/index.ts:5]: I'd suggest moving duplicate logic (like error handling, response formatting) into separate lib/ functions and use them across the handlers.

It's a wish for the future, no need to implement it right now. It won't affect your score.

File: product-service/handlers/get-product-by-id/index.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+    return {
+      statusCode: 200,
+      body: JSON.stringify(product),
+    };
+
+  }catch (error) {
+    console.error(error);
+    return {
+      statusCode: 400,
```
Comment (Emerkof) [inline product-service/handlers/get-product-by-id/index.ts:24]: I'd suggest implementing an errorHandler function, which would process varoius types of error. Currently, all the errors thrown will send the `400` statusCode, which is not correct (for example, if a product is not found, the `statusCode` should be `404`).

It's a wish for the future, no need to implement it right now. It won't affect your score.

**Comment by artsiompeshko** [comment]
Your FE integration is not completed, you just updated the API path, but there also should be updates inside components to fetch products with API. 

I believe you just missed it, so I will anyway set a maximum 11 mark, but pls finish FE updates.

**Review by chermic — APPROVED** [review]

**Pull request merged**